components/common/mojomCache.ts

```diff
@@ -45,3 +37,29 @@ export class CachingWrapper<Entity> {
     }
   }
 }
+
+export type Cache<Entity> = { [key: string]: Entity }
+export type ChangeEvent<Entity> = {
+  addedOrUpdated: { [key: string]: Entity }
+  removed: string[]
+}
+
+/**
+ * A helper class for caching a dictionary of entities which may be
+ * added/updated/remove.
+ *
+ * Manages partial updates and notifies listeners of the changes.
+ */
+export class EntityCachingWrapper<Entity> extends CachingWrapper<Cache<Entity>> {
+  constructor() {
+    super({})
+  }
+
+  changed (event: ChangeEvent<Entity>) {
```

Comment:
comment this function so we know it gets called from C++ / mojom implementation (if that's right?)